test/tests/amqp/test_amqp_node_rediscover.py

```diff
@@ -0,0 +1,537 @@
+        try:
+            amqp_body_json = json.loads(amqp_body)
+        except ValueError:
+            self.fail("FAILURE - The message body is not json format!")
+        self.assertIn("nodeId", amqp_body_json["data"], "nodeId is not contained in the discover message")
+        self.assertNotEquals(
+            amqp_body_json["data"]["nodeId"], "", "nodeId generated in discovery doesn't include valid data ")
+        self.assertIn(
+            "ipMacAddresses", amqp_body_json["data"], "ipMacAddresses is not contained in the discover message")
+        self.assertNotEquals(
+            amqp_body_json["data"]["ipMacAddresses"], "",
+            "ipMacAddresses generated during node discovery doesn't include valid data ")
+
+    def _wait_for_uuid(self, node_uuid):
+        for dummy in range(0, 20):
+            sleep(30)
+            rest_data = fit_common.rackhdapi('/redfish/v1/Systems/')
+            if rest_data['json']['Members@odata.count'] == 0:
+                continue
+            node_collection = rest_data['json']['Members']
+            for computenode in node_collection:
+                nodeidurl = computenode['@odata.id']
+                api_data = fit_common.rackhdapi(nodeidurl)
+                if api_data['status'] > 399:
+                    break
+                if node_uuid == api_data['json']['UUID']:
+                    return True
+        logs.debug_3("Time out to find the node with uuid!")
+        return False
+
+    def _wait_amqp_message(self, timeout):
+        timecount = 0
+        while amqp_queue.empty() is True and timecount < timeout:
+            sleep(1)
+            timecount = timecount + 1
+        self.assertNotEquals(
+            timecount,
+            timeout,
+            "AMQP message receive timeout")
+
+    def amqp_callback(self, ch, method, properties, body):
+        logs.debug_3("Routing Key {0}:".format(method.routing_key))
+        logs.data_log.debug_3(body.__str__())
+        global amqp_queue, nodefound_id
+        amqp_queue.put(
+            [method.routing_key, fit_common.json.loads(body)["nodeId"], body])
+        nodefound_id = fit_common.json.loads(body)["nodeId"]
+
+    def _check_skupack(self):
+        sku_installed = fit_common.rackhdapi('/api/2.0/skus')['json']
+        if len(sku_installed) < 2:
+            return False
+        else:
+            return True
+
+    def _get_tester_ip(self):
+        s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        ip = fit_common.fitargs()['rackhd_host']
+        logs.debug("pinging " + ip)
+        s.connect((ip, 0))
+        logs.debug('My IP address is: ' + s.getsockname()[0])
+        return str(s.getsockname()[0])
+
+    def _process_web_message(self, timeout):
+        global webhook_received, webhook_body
+        timecount = 0
+        while webhook_received is False and timecount < timeout:
+            sleep(1)
+            timecount = timecount + 1
+        self.assertNotEquals(

```

Comment:
don't need line break, it's not easy to code review for these unnecessary line breaks